core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java

```diff
@@ -89,8 +90,11 @@ private Builder() {
     }
 
     public Builder withTableMetadata(TableMetadata tableMetadata) {
-      this.metadataLocation = tableMetadata.metadataFileLocation();
       this.metadata = tableMetadata;
+      if (tableMetadata.metadataFileLocation() != null) {
+        this.metadataLocation = tableMetadata.metadataFileLocation();
+      }
```

Comment:
As we've removed the `withMetadataLocation` method from the builder, should I revert this change?

I do like that it shows the nullability a bit further to the reader, but it is arguably an unnecessary change.

Reply:
Yes, please remove unnecessary changes.

Reply:
This has been reverted.